### ISSUE TITLE
[salt-cloud saltify]  AttributeError: 'str' object has no attribute 'setdefault'

### ISSUE THREAD

**Issue by arthurlogilab** [saltstack/salt#24021]
While failing on #24020  I tried a map version of the same thing and go this : 

```
# salt-cloud -P -m map_saltify 
[INFO    ] salt-cloud starting
ursa:
    True
[INFO    ] Runner completed: 20150521122436271395
[ERROR   ] An un-handled exception was caught by salt's global exception handler:
AttributeError: 'str' object has no attribute 'setdefault'
Traceback (most recent call last):
  File "/usr/bin/salt-cloud", line 10, in <module>
    salt_cloud()
  File "/usr/lib/python2.7/dist-packages/salt/scripts.py", line 316, in salt_cloud
    client.run()
  File "/usr/lib/python2.7/dist-packages/salt/cloud/cli.py", line 86, in run
    mapper = salt.cloud.Map(self.config)
  File "/usr/lib/python2.7/dist-packages/salt/cloud/__init__.py", line 1558, in __init__
    self.rendered_map = self.read()
  File "/usr/lib/python2.7/dist-packages/salt/cloud/__init__.py", line 1685, in read
    overrides.setdefault('name', name)
AttributeError: 'str' object has no attribute 'setdefault'
Traceback (most recent call last):
  File "/usr/bin/salt-cloud", line 10, in <module>
    salt_cloud()
  File "/usr/lib/python2.7/dist-packages/salt/scripts.py", line 316, in salt_cloud
    client.run()
  File "/usr/lib/python2.7/dist-packages/salt/cloud/cli.py", line 86, in run
    mapper = salt.cloud.Map(self.config)
  File "/usr/lib/python2.7/dist-packages/salt/cloud/__init__.py", line 1558, in __init__
    self.rendered_map = self.read()
  File "/usr/lib/python2.7/dist-packages/salt/cloud/__init__.py", line 1685, in read
    overrides.setdefault('name', name)
AttributeError: 'str' object has no attribute 'setdefault'
# salt-cloud --versions-report
                  Salt: 2015.5.0
                Python: 2.7.6 (default, Mar 22 2014, 22:59:56)
                Jinja2: 2.7.2
              M2Crypto: 0.21.1
        msgpack-python: 0.3.0
          msgpack-pure: Not Installed
              pycrypto: 2.6.1
               libnacl: Not Installed
                PyYAML: 3.10
                 ioflo: Not Installed
                 PyZMQ: 14.0.1
                  RAET: Not Installed
                   ZMQ: 4.0.4
                  Mako: 0.9.1
       Apache Libcloud: 0.15.1
 Debian source package: 2015.5.0+ds-1trusty1
# more map_saltify 
make_salty:
  - name:
    ssh_host: 172.17.17.28
    ssh_username: debian
    ssh_keyfile: '/etc/salt/ursa-salt-key.pem'
    sudo: True
```

**Comment by marek-obuchowicz** [comment]
I had the same problem when tried to use my old map files (they used to work one year ago), which looked like:
```
ec2_eu_qa_web_t2_small_1a:
  - staging-web01:
    grains:
      roles:
        - web
```

Apparently this syntax worked:
```
ec2_eu_qa_web_t2_small_1a:
  - staging-web05:
      grains:
        - roles: web
```

**Comment by rallytime** [comment]
@arthurlogilab If you bump the information below `- name:` over two spaces, do you still see the same error? Like so:
```
make_salty:
  - name:
      ssh_host: 172.17.17.28
      ssh_username: debian
      ssh_keyfile: '/etc/salt/ursa-salt-key.pem'
      sudo: True
```
I am also wondering if you should change the title of `name` to something else. `name` may be a "reserved word" in this case. 

If that fixes the bug, I'll edit the docs so people don't run into this that again.

**Comment by rallytime** [comment]
@arthurlogilab - I just tested this by changing my map file to match something similar to yours and I was able to reproduce the error at the HEAD of 2015.5:
```
# salt-cloud -m ~/mapfile
[INFO    ] salt-cloud starting
[ERROR   ] An un-handled exception was caught by salt's global exception handler:
AttributeError: 'str' object has no attribute 'setdefault'
Traceback (most recent call last):
  File "/usr/local/bin/salt-cloud", line 10, in <module>
    salt_cloud()
  File "/root/SaltStack/salt/salt/scripts.py", line 316, in salt_cloud
    client.run()
  File "/root/SaltStack/salt/salt/cloud/cli.py", line 87, in run
    mapper = salt.cloud.Map(self.config)
  File "/root/SaltStack/salt/salt/cloud/__init__.py", line 1565, in __init__
    self.rendered_map = self.read()
  File "/root/SaltStack/salt/salt/cloud/__init__.py", line 1699, in read
    overrides.setdefault('name', name)
AttributeError: 'str' object has no attribute 'setdefault'
Traceback (most recent call last):
  File "/usr/local/bin/salt-cloud", line 10, in <module>
    salt_cloud()
  File "/root/SaltStack/salt/salt/scripts.py", line 316, in salt_cloud
    client.run()
  File "/root/SaltStack/salt/salt/cloud/cli.py", line 87, in run
    mapper = salt.cloud.Map(self.config)
  File "/root/SaltStack/salt/salt/cloud/__init__.py", line 1565, in __init__
    self.rendered_map = self.read()
  File "/root/SaltStack/salt/salt/cloud/__init__.py", line 1699, in read
    overrides.setdefault('name', name)
AttributeError: 'str' object has no attribute 'setdefault'
```
I'll clean up that error in a pull request shortly and make an error message to help out in this situation in the future.

That being said, if you change your configs to the ones I suggested above (changing the `name` param, and adding two spaces for the `ssh_user` information, etc, and also note the change from `ssh_keyfile` to `key_filename`), I have tested this functionality based on the new docs I submitted in #26790 and it works very well.

**Comment by rallytime** [comment]
I've fixed the stack trace with #26793. Instead of the stacktrace, the following message will be shown instead:
```
# salt-cloud -m /etc/salt/mapfile
[INFO    ] salt-cloud starting
[ERROR   ] Cannot use 'name' as a minion id in a cloud map as it is a reserved word. Please change 'name' to a different minion id reference.
```
